pytorch_lightning/trainer/training_tricks.py

```diff
@@ -281,14 +282,26 @@ def _adjust_batch_size(trainer,
 
         desc: either `succeeded` or `failed`. Used purely for logging
 
+    Returns:
+        The new batch size for the next trial and a bool that signals whether the
+        new value is different than the previous batch size.
     """
     model = trainer.get_model()
     batch_size = lightning_getattr(model, batch_arg_name)
     new_size = value if value is not None else int(batch_size * factor)
     if desc:
         log.info(f'Batch size {batch_size} {desc}, trying batch size {new_size}')
+
+    if not _is_valid_batch_size(new_size, trainer.train_dataloader):
```

Comment:
This function is only used here, so we could just write the expression directly here.

Reply:
I made it not with the primary goal for reuse, but more to describe what it does with the method name, but granted, maybe that is not much help.